core/src/main/java/com/linecorp/armeria/common/multipart/MimeParser.java

```diff
@@ -241,13 +245,22 @@ void parse() {
                         final ByteBuf bodyContent = readBody();
                         if (boundaryStart == -1 || bodyContent == NEED_MORE) {
                             if (bodyContent == NEED_MORE) {
-                                // Need more data
+                                final BodyPartPublisher currentPublisher = bodyPartPublisher;
+                                currentPublisher.whenConsumed().thenRun(() -> {
+                                    if (currentPublisher.demand() > 0 && !currentPublisher.isComplete()) {
+                                        multipartDecoder.requestUpstreamForBodyPartData();
+                                    }
+                                });
                                 return;
                             }
                         } else {
                             startOfLine = false;
                         }
-                        bodyPartPublisher.write(HttpData.wrap(bodyContent));
+
+                        // Use tryWrite() to avoid throwing exception.
+                        // For example, when body part is cancelled, MimeParser need to ignore it without
+                        // throwing exception.
+                        bodyPartPublisher.tryWrite(HttpData.wrap(bodyContent));
```

Comment:
`tryWrite` will handle the case like body is cancelled.
Due to even body is cancelled, the upstream will still sending the httpdata until next BodyPart starts. So here just ignore it to like we want the data, but just ignore it.

If we make state fails, it will handle whole multipart as failed IIUC.